src/main/java/org/candlepin/resource/CloudRegistrationResource.java

```diff
@@ -68,12 +72,28 @@ public String cloudAuthorize(CloudRegistrationDTO cloudRegistrationDTO) {
             String errmsg = this.i18n.tr("Cloud registration is not supported by this Candlepin instance");
             throw new NotImplementedException(errmsg, e);
         }
-        catch (CloudRegistrationAuthorizationException e) {
-            throw new NotAuthorizedException(e.getMessage());
-        }
-        catch (MalformedCloudRegistrationException e) {
-            throw new BadRequestException(e.getMessage(), e);
+        catch (CloudRegistrationServiceException e) {
+            if (e instanceof CloudRegistrationAuthorizationException) {
+                throw new NotAuthorizedException(e.getMessage());
+            }
+            else if (e instanceof CloudRegistrationMissingTypeException) {
+                throw new BadRequestException(this.i18n.tr(
+                    "Request is missing cloud provider type (e.g. amazon, gcp, azure)"));
+            }
+            else if (e instanceof CloudRegistrationBadTypeException) {
```

Comment:
One more thought. Rather than catching and mapping, we could provide hosted with a set of exceptions that have the translation integrated inside. That way we wouldn't need to do any mapping on our side. All hosted exceptions would be translated. E.G.

```java
public class CloudRegistrationBadTypeException extends ProductServiceException {
    public CloudRegistrationBadTypeException(I18n i18n, String type) {
        super(i18n.tr("Unrecognized provider type in request: {}", type));
    }
}
```